index/scorch/segment/zap/contentcoder_test.go

```diff
@@ -73,3 +78,49 @@ func TestChunkContentCoder(t *testing.T) {
 		}
 	}
 }
+
+func TestChunkedContentCoders(t *testing.T) {
+	maxDocNum := uint64(5)
+	chunkSize := uint64(1)
+	docNums := []uint64{0, 1, 2, 3, 4, 5}
+	vals := [][]byte{
+		[]byte("scorch"),
+		[]byte("does"),
+		[]byte("better"),
+		[]byte("than"),
+		[]byte("upside"),
+		[]byte("down"),
+	}
+
+	var actual1, actual2 bytes.Buffer
+	// chunkedContentCoder that writes out at the end
+	cic1 := newChunkedContentCoder(chunkSize, maxDocNum, &actual1, false)
+	// chunkedContentCoder that writes out in chunks
+	cic2 := newChunkedContentCoder(chunkSize, maxDocNum, &actual2, true)
+
+	for i, docNum := range docNums {
+		err := cic1.Add(docNum, vals[i])
+		if err != nil {
+			t.Fatalf("error adding to intcoder: %v", err)
+		}
+		err = cic2.Add(docNum, vals[i])
+		if err != nil {
+			t.Fatalf("error adding to intcoder: %v", err)
+		}
+	}
+	_ = cic1.Close()
+	_ = cic2.Close()
+
+	_, err := cic1.Write()
+	if err != nil {
+		t.Fatalf("error writing: %v", err)
+	}
+	_, err = cic2.Write()
+	if err != nil {
+		t.Fatalf("error writing: %v", err)
+	}
+
+	if !reflect.DeepEqual(string(actual1.Bytes()), string(actual2.Bytes())) {
```

Comment:
can use bytes.Compare() instead of DeepEqual()

Reply:
Done.